survey/src/main/kotlin/com/surveyor/survey/controllers/SurveyController.kt

```diff
@@ -0,0 +1,59 @@
+package com.surveyor.survey.controllers
+
+import com.surveyor.survey.services.QuestionService
+import com.surveyor.survey.services.SurveyService
+import java.util.*
+import org.springframework.http.ResponseEntity
+import org.springframework.http.HttpStatus
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.bind.annotation.*
+import org.springframework.validation.BindingResult
+import javax.validation.Valid 
+import com.surveyor.survey.models.SurveyRequest
+import com.surveyor.survey.models.SurveyDTO
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.surveyor.survey.entities.Answer
+import com.surveyor.survey.mappers.SurveyMapper
+
+
+@RestController
+@RequestMapping("/survey")
+class SurveyController(private val surveyService: SurveyService,
+private val surveyMapper: SurveyMapper) {
+    @GetMapping("/{id}")
+    fun getSurveyResponse(@PathVariable id: Long): ResponseEntity<String>{
+        var responsesDTO: SurveyDTO;
+        try {
+            var survey = this.surveyService.findById(id);
+            responsesDTO = this.surveyMapper.toSurveyDTO(survey);
+        }
+        catch(e: Exception) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Bad Request");
+        }
+        return ResponseEntity.status(HttpStatus.OK).body(this.surveyMapper.toJsonString(responsesDTO));
+    }
+
+    @PostMapping("/{id}")
```

Comment:
```suggestion
    @PostMapping("/{id}/responses")
```

To be [RESTful](https://restfulapi.net/rest-api-design-tutorial-with-example/), It's better to have it like this, Do you have another convention you follow?